src/plugins/controls/public/options_list/options_list_reducers.ts

```diff
@@ -38,7 +43,16 @@ export const optionsListReducers = {
     }
   },
   setSearchString: (state: WritableDraft<OptionsListReduxState>, action: PayloadAction<string>) => {
-    state.componentState.searchString = action.payload;
+    state.componentState.searchString.value = action.payload;
+    if (
+      action.payload !== '' && // empty string search is never invalid
+      state.componentState.field?.type === 'ip' && // only IP searches can currently be invalid
+      !getIpRangeQuery(action.payload).validSearch
```

Comment:
Ahh great point! 👍 Definitely agree we could make this more universal once we introduce another field type that requires separate validation - otherwise, this logic could get pretty ugly.